File: util/wechat/src/main/java/cn/bestwu/summer/util/wechat/config/WechatConfiguration.java
package cn.bestwu.summer.util.wechat.config;

import cn.bestwu.summer.util.wechat.controller.WechatCallbackController;
import cn.bestwu.summer.util.wechat.support.IWechatService;
import com.riversoft.weixin.mp.base.AppSetting;
import com.riversoft.weixin.mp.oauth2.MpOAuth2s;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@ConditionalOnClass(AppSetting.class)
@ConditionalOnProperty(prefix = "wechat", name = "appId")
@Configuration
@EnableConfigurationProperties({WechatProperties.class})
public class WechatConfiguration {

  private final Logger log = LoggerFactory.getLogger(WechatConfiguration.class);
  private final WechatProperties wechatProperties;

  public WechatConfiguration(WechatProperties wechatProperties) {
    this.wechatProperties = wechatProperties;
    try {
      String authenticationUrl = MpOAuth2s.with(wechatProperties)
          .authenticationUrl(wechatProperties.getOauthUrl(), "snsapi_userinfo");
      log.info("authenticationUrl:{}", authenticationUrl);
    } catch (Exception e) {
      log.error(e.getMessage(), e);
    }
  }

  @Bean
  public WechatCallbackController wechatCallbackController(IWechatService wechatService) {
    return new WechatCallbackController(wechatService, wechatProperties);
  }

}

File: util/weixin/src/main/kotlin/top/bettercode/summer/util/wechat/config/WechatConfiguration.kt
package top.bettercode.summer.util.wechat.config

import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import top.bettercode.summer.util.wechat.controller.OffiaccountCallbackController
import top.bettercode.summer.util.wechat.support.IWechatService
import top.bettercode.summer.util.wechat.support.offiaccount.IOffiaccountClient
import top.bettercode.summer.util.wechat.support.offiaccount.OffiaccountClient

@Configuration(proxyBeanMethods = false)
@ConditionalOnBean(IOffiaccountProperties::class)
class WechatConfiguration(private val offiaccountProperties: IOffiaccountProperties) {

    @Bean
    fun offiaccountClient(): IOffiaccountClient {
        return OffiaccountClient(offiaccountProperties)
    }

    @ConditionalOnWebApplication
    @Bean
    fun offiaccountCallbackController(
        wechatService: IWechatService,
        offiaccountClient: IOffiaccountClient
    ): OffiaccountCallbackController {
        return OffiaccountCallbackController(wechatService, offiaccountClient)
    }

}